app/controlplane/plugins/plugins.go

```diff
@@ -44,33 +74,144 @@ func Load(l log.Logger) (sdk.AvailablePlugins, error) {
 		slack.New,
```

Comment:
> Does it make sense to export "optional" plugins to different repositories outside "chainloop" repository?
It is a good way of teaching good practices to the community to ease contributions.

Indeed, that's the direction, specially to showcase "bring your own plugin" kind of thing. Out of curiosity, why would you say that having the plugins code in third party repositories is a better practice?

> Later, you can configure your chainloop image to include pre-compiled binaries from GitHub releases (for each of those new plugin repositories) or a similar approach to place them in plugins/bin. That makes the usability experience self-documented.

Yes and yes :)

> For example, I'd probably keep the ociregistry plugin as built-in as I understand is kind of a strict dependency even though it is pluggable and replaceable given the lazy-load mechanism (built-in plugins are not loaded if they are superseded by pre-compiled plugins with the same name).

Good point, although first, we need to figure out the matrix terms, for example, we have two axis, "is it plugin based" and "is it provided/bundled by chainloop"

On the is it plugins based. This probably should be the default and only option in the future, so even OCIregistry should use this mechanism, I think. 

the "is it provided by chainloop" is where it will probably define where the source code will be and hence your previous points. I definitely see a future of third party plugins on their own repo.

Reply:
hi @Javirln, thanks for the pointers, it's super useful :)

Let me try to get it right by summarizing it. In your option, you'd rather have plugins (core and contrib) in another repo, even if it's a monorepo of extensions (i.e chainloop-contrib).

You think we should have a single way of doing things, meaning all plugins or nothing, regardless of their importance or status (core or contrib). 

Those make sense to me!